pkgs/by-name/cb/cbtemulator/package.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs
+, lib
+, stdenv
+, buildGoModule
+, fetchFromGitHub
+, runCommand
+, google-cloud-bigtable-tool
+, procps
+}:
+
+buildGoModule rec {
+  pname = "cbtemulator";
+  version = "1.22.0";
+
+  src = (fetchFromGitHub {
+    owner = "googleapis";
+    repo = "google-cloud-go";
+    rev = "bigtable/v${version}";
+    hash = "sha256-eOi4QFthnmZb5ry/5L7wzr4Fy1pF/H07BzxOnXtmSu4=";
+  }) + "/bigtable";
+
+  vendorHash = "sha256-7M7YZfl0usjN9hLGozqJV2bGh+M1ec4PZRGYUhEckpY=";
+  subPackages = [ "cmd/emulator" ];
+
+  postInstall = ''
+    mv $out/bin/emulator $out/bin/cbtemulator
+  '';
+
+  passthru = {
+    # Sets up a table and family, then inserts, and ensures it gets back the value.
+    tests.smoketest = runCommand "cbtemulator-smoketest"
+      {
+        nativeBuildInputs = [ google-cloud-bigtable-tool procps ];
+      } ''
+      # Start the emulator
+      ${lib.getExe pkgs.cbtemulator} &
+
+      cleanup() {
+        pkill cbtemulator
```

Comment:
Note to self: this doesn't seem to bring pkill into $PATH, check Darwin manually

Reply:
`procps` doesn't have `pkill` on MacOS, but I rewrote this to just use the builtin `kill` bash builtin.